core/src/apps/ethereum/yielding.py

```
@@ -135,22 +206,214 @@ def _prepare_vault_tx(
+                log.debug(
+                    "SS DEBUG",
+                    "_prepare_claim_rewards: token[%d] addr not bytes, raising InvalidFunctionCall",
+                    i,
+                )
+                raise InvalidFunctionCall
+            label = get_token_label(addr, network)
+            log.debug("SS DEBUG", "token[%d] addr=%s label=%s", i, addr, label)
+            token_labels.append(label)
+
+    except (ValueError, EOFError, InvalidFunctionCall):
+        raise DataError("Invalid data for claim rewards transaction")
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)